src/librustc/lint/mod.rs

```diff
@@ -126,46 +128,92 @@ pub trait LintPass {
+    fn check_stmt(&mut self, _: &LateContext, _: &hir::Stmt) { }
+    fn check_arm(&mut self, _: &LateContext, _: &hir::Arm) { }
+    fn check_pat(&mut self, _: &LateContext, _: &hir::Pat) { }
+    fn check_decl(&mut self, _: &LateContext, _: &hir::Decl) { }
+    fn check_expr(&mut self, _: &LateContext, _: &hir::Expr) { }
+    fn check_expr_post(&mut self, _: &LateContext, _: &hir::Expr) { }
+    fn check_ty(&mut self, _: &LateContext, _: &hir::Ty) { }
+    fn check_generics(&mut self, _: &LateContext, _: &hir::Generics) { }
+    fn check_fn(&mut self, _: &LateContext,
         _: FnKind, _: &hir::FnDecl, _: &hir::Block, _: Span, _: ast::NodeId) { }
-    fn check_trait_item(&mut self, _: &Context, _: &hir::TraitItem) { }
-    fn check_impl_item(&mut self, _: &Context, _: &hir::ImplItem) { }
-    fn check_struct_def(&mut self, _: &Context,
+    fn check_trait_item(&mut self, _: &LateContext, _: &hir::TraitItem) { }
+    fn check_impl_item(&mut self, _: &LateContext, _: &hir::ImplItem) { }
+    fn check_struct_def(&mut self, _: &LateContext,
         _: &hir::StructDef, _: ast::Ident, _: &hir::Generics, _: ast::NodeId) { }
-    fn check_struct_def_post(&mut self, _: &Context,
+    fn check_struct_def_post(&mut self, _: &LateContext,
         _: &hir::StructDef, _: ast::Ident, _: &hir::Generics, _: ast::NodeId) { }
-    fn check_struct_field(&mut self, _: &Context, _: &hir::StructField) { }
-    fn check_variant(&mut self, _: &Context, _: &hir::Variant, _: &hir::Generics) { }
-    fn check_variant_post(&mut self, _: &Context, _: &hir::Variant, _: &hir::Generics) { }
-    fn check_opt_lifetime_ref(&mut self, _: &Context, _: Span, _: &Option<hir::Lifetime>) { }
-    fn check_lifetime_ref(&mut self, _: &Context, _: &hir::Lifetime) { }
-    fn check_lifetime_def(&mut self, _: &Context, _: &hir::LifetimeDef) { }
-    fn check_explicit_self(&mut self, _: &Context, _: &hir::ExplicitSelf) { }
-    fn check_mac(&mut self, _: &Context, _: &ast::Mac) { }
-    fn check_path(&mut self, _: &Context, _: &hir::Path, _: ast::NodeId) { }
-    fn check_attribute(&mut self, _: &Context, _: &hir::Attribute) { }
+    fn check_struct_field(&mut self, _: &LateContext, _: &hir::StructField) { }
+    fn check_variant(&mut self, _: &LateContext, _: &hir::Variant, _: &hir::Generics) { }
+    fn check_variant_post(&mut self, _: &LateContext, _: &hir::Variant, _: &hir::Generics) { }
+    fn check_opt_lifetime_ref(&mut self, _: &LateContext, _: Span, _: &Option<hir::Lifetime>) { }
+    fn check_lifetime_ref(&mut self, _: &LateContext, _: &hir::Lifetime) { }
+    fn check_lifetime_def(&mut self, _: &LateContext, _: &hir::LifetimeDef) { }
+    fn check_explicit_self(&mut self, _: &LateContext, _: &hir::ExplicitSelf) { }
+    // Note that you shouldn't implement both check_mac and check_ast_mac,
+    // because then your lint will be called twice. Prefer check_ast_mac.
+    fn check_mac(&mut self, _: &LateContext, _: &ast::Mac) { }
```

Comment:
@Manishearth 

> Why would we have a macro in the HIR tree? Isn't HIR post-expansion?

I believe these are exported macro definitions, not macro uses.